gridstatus/ieso.py

```diff
@@ -844,3 +910,788 @@ def _request(self, url, verbose):
             )
 
         return r
+
+    @support_date_range(frequency="DAY_START")
+    def get_resource_adequacy_report(
+        self,
+        date: str | datetime.date | datetime.datetime,
+        end: datetime.date | datetime.datetime | None = None,
+        vintage: Literal["all", "latest"] = "latest",
+        last_modified: str | datetime.date | datetime.datetime | None = None,
```

Comment:
Do you think we should add `Last Modified` as a column to the returned dataframe for auditing? Might be helpful to have if there are issues.

Reply:
Yeah I think it could help. There is also an issue with the `vintage="all"` in that the unversioned file and the highest versioned file contain the same data, which results in duplicates. 

I'm thinking for all vintages leaving behind the unversioned file, with the knowledge that the versioned files contain all the unique data.

Reply:
Conceivably a `drop_duplicates()` at the end of the processing would also work

Reply:
added `last_modified` as a column to the returned df.

Reply:
Okay, added last modified as a column and will do a drop duplicates in places that need it.

Reply:
Sorry, what I meant for "Last Modified" was the value associated with the file from this page: https://reports-public.ieso.ca/public/Adequacy2/

Reply:
Ah. 

Okay I've implemented this, I believe it's working correctly.